### PR TITLE
Observation Plans submission

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR enables telescope queue submissions for the ObservationPlanRequests.

### CODE REVIEW

File: skyportal/models/instrument.py [hunk 1]
```diff
@@ -180,6 +180,18 @@ class InstrumentField(Base):
         nullable=False,
     )
 
+    ra = sa.Column(
+        sa.Float,
+        doc='The mid-point right ascension for the tile.',
```
Comment (guynir42) [inline skyportal/models/instrument.py:185]: Docstring should mention the units. Same for dec.

**Review by guynir42 — APPROVED** [review]
I have only minor comments. After that it is good to go.

File: skyportal/facility_apis/ztf.py [hunk 2]
```diff
@@ -236,6 +295,115 @@ class ZTFMMAAPI(MMAAPI):
...
+            raise ValueError('Missing allocation information.')
+
+        queue_name = "ToO_" + request.payload["queue_name"]
+        headers = {"Authorization": f"Bearer {altdata['access_token']}"}
+
+        payload = {'queue_name': queue_name, 'user': request.requester.username}
+
+        url = urllib.parse.urljoin(ZTF_URL, 'api/triggers/ztf')
+        s = Session()
+        req = Request('DELETE', url, json=payload, headers=headers)
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/facility_apis/ztf.py:391]: Yes definitely not good!

File: skyportal/models/instrument.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/models/instrument.py:185]: yes.

File: skyportal/facility_apis/ztf.py [hunk 3]
```diff
@@ -236,6 +295,115 @@ class ZTFMMAAPI(MMAAPI):
...
+            'queue_type': 'list',
+            'user': request.requester.username,
+        }
+
+        url = urllib.parse.urljoin(ZTF_URL, 'api/triggers/ztf')
+        s = Session()
+        req = Request('PUT', url, json=payload, headers=headers)
+        prepped = req.prepare()
+        r = s.send(prepped)
+        print(r.text)
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/facility_apis/ztf.py:334]: yup.

File: skyportal/facility_apis/ztf.py [hunk 4]
```diff
@@ -236,6 +295,115 @@ class ZTFMMAAPI(MMAAPI):
...
+        # r.raise_for_status()
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/facility_apis/ztf.py:335]: yup.

File: skyportal/facility_apis/ztf.py [hunk 5]
```diff
@@ -77,6 +77,65 @@ def _build_payload(self, request):
...
+        }
+
+        if request.payload["program_id"] == "Partnership":
+            program_id = 2
+        elif request.payload["program_id"] == "Caltech":
+            program_id = 3
+        else:
+            raise ValueError('Unknown program.')
+
+        observation_plan = request.observation_plans[0]
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/facility_apis/ztf.py:111]: I am adding a check that there is only one observation plan.

**Comment by mcoughlin** [comment]
Failures unrelated. May I merge @guynir42 ?

**Review by guynir42 — APPROVED** [review]
You can merge.

**Pull request merged**